awscli/botocore/utils.py

```diff
@@ -3980,3 +3980,11 @@ def is_s3express_bucket(bucket):
     'stepfunctions': 'sfn',
     'storagegateway': 'storage-gateway',
 }
+
+
+class DefaultTrue:
+    def __eq__(self, other):
+        return other == True
```

Comment:
1 minor concern with this sentinel is that it doesn't work for identity operators (eg `DefaultTrue() is True` evaluates to `False`).

This is also a workaround to compensate for resolving configured values and I'm not sure we should be legitimizing it and potentially encouraging future contributors to use it.

What do you think of @ashovlin's suggestion of storing a public and private `inject_host_prefix` attribute in the `Config` class? We would have to compare against a private attribute in `args.py`, but it seems marginally better than replacing the default with this sentinel value.